src/masternode-sync.cpp

```diff
@@ -74,43 +83,36 @@ void CMasternodeSync::Process()
         */
         if(mnodeman.CountEnabled() == 0) {
             RequestedMasternodeAssets = MASTERNODE_SYNC_INITIAL;
-            RequestedMasternodeAttempt = 0;
-        }
-        return;
+        } else
+            return;
     }
 
-    if(c++ % MASTERNODE_SYNC_TIMEOUT != 0) return;
+    if(fDebug) LogPrintf("CMasternodeSync::Process() - tick %d RequestedMasternodeAssets %d\n", tick, RequestedMasternodeAssets);
 
-    if(fDebug) LogPrintf("CMasternodeSync::Process() - RequestedMasternodeAssets %d c %d\n", RequestedMasternodeAssets, c);
-
-    //request full mn list only if Masternodes.dat was updated quite a long time ago
     if(RequestedMasternodeAssets == MASTERNODE_SYNC_INITIAL) GetNextAsset();
 
-    CBlockIndex* pindexPrev = chainActive.Tip();
-    if(pindexPrev == NULL) return;
-
     LOCK(cs_vNodes);
     BOOST_FOREACH(CNode* pnode, vNodes)
     {
-        if (pnode->nVersion >= MIN_POOL_PEER_PROTO_VERSION)
-        {
-
-            if(RequestedMasternodeAssets == MASTERNODE_SYNC_SPORKS){
-                if(pnode->HasFulfilledRequest("getspork")) continue;
-                pnode->FulfilledRequest("getspork");
-
-                if(RequestedMasternodeAttempt <= 2){
-                    pnode->PushMessage("getsporks"); //get current network sporks
-                    if(RequestedMasternodeAttempt == 2) GetNextAsset();
-                    RequestedMasternodeAttempt++;
-                }
-                return;
+        if(RequestedMasternodeAssets == MASTERNODE_SYNC_SPORKS){
+            if(pnode->HasFulfilledRequest("getspork")) continue;
+            pnode->FulfilledRequest("getspork");
+
+            if(RequestedMasternodeAttempt <= 2){
+                pnode->PushMessage("getsporks"); //get current network sporks
+                if(RequestedMasternodeAttempt == 2) GetNextAsset();
+                RequestedMasternodeAttempt++;
             }
+            return;
+        }
 
-            //don't begin syncing until we're at a recent block
-            if(pindexPrev->nTime + 600 < GetTime()) return;
+        //don't begin syncing until we're almost at a recent block
+        if(pindexPrev->nHeight + 4 < pindexBestHeader->nHeight || pindexPrev->nTime + 600 < GetTime()) return;
```

Comment:
Shouldn't we wait to ask for mnb/mbp until we have the most recent block? Otherwise if a masternodes input is newer than the current block, we will reject it and have inconsistancies